tests/runner/test_storage.py

```diff
@@ -136,6 +162,286 @@ def test_repr(self):
         repo = GitRepository(url="https://github.com/org/repo.git")
         assert (
             repr(repo)
-            == "GitRepository(name='repo-main'"
-            " repository='https://github.com/org/repo.git', branch='main')"
+            == "GitRepository(name='repo'"
+            " repository='https://github.com/org/repo.git', branch=None)"
+        )
+
+    async def test_include_submodules_property(
+        self, mock_run_process: AsyncMock, monkeypatch
+    ):
+        repo = GitRepository(
+            url="https://github.com/org/repo.git", include_submodules=True
+        )
+        await repo.pull_code()
+        mock_run_process.assert_awaited_with(
+            [
+                "git",
+                "clone",
+                "https://github.com/org/repo.git",
+                "--recurse-submodules",
+                "--depth",
+                "1",
+                str(Path.cwd() / "repo"),
+            ]
+        )
+
+        # pretend the repo already exists
+        monkeypatch.setattr("pathlib.Path.exists", lambda x: ".git" in str(x))
+
+        await repo.pull_code()
+        mock_run_process.assert_awaited_with(
+            [
+                "git",
+                "pull",
+                "origin",
+                "--recurse-submodules",
+                "--depth",
+                "1",
+            ],
+            cwd=Path.cwd() / "repo",
+        )
+
+    async def test_git_clone_errors_obscure_access_token(self, monkeypatch):
```

Comment:
I think it would be good to test that credentials are not in stdout/stderr in the case that there's an auth error against the git URL.

Reply:
Added an assertion with `capsys` in [c65dfc8](https://github.com/PrefectHQ/prefect/pull/10957/commits/c65dfc81189967cc5dc4257fde8931d868ce9613)